src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/RouteAttributeExtensionNode.cs

```diff
@@ -32,6 +32,12 @@ public override void WriteNode(CodeTarget target, CodeRenderingContext context)
                 context.CodeWriter.WriteLine(Template);
             }
         }
+
+        if (isStaticPage)
+        {
+            context.CodeWriter.Write(", Static = true");
```

Comment:
This won't work in .NET < 9, right? So it should be gated by RazorLanguageVersion, reporting an error for `@staticpage` in earlier razor versions.

Reply:
I covered this in the remarks about "Limiting to 9.0+" in the PR description above.

Reply:
Sorry, I've missed that. `RazorLanguageVersion.Version_9_0` already exists in `main`, it's been added in https://github.com/dotnet/razor/pull/10150. I don't think net9.0 is needed for that.

Reply:
> Nothing immediately bad will happen if the directive lights up on 8.0

I think that's not true, this is a valid razor in .NET 8:

```razor
@staticpage

@code {
    private string staticpage = "test";
}
```

Adding any Razor directive seems like a breaking change in this regard (but probably one we tolerate).

Reply:
> Adding any Razor directive seems like a breaking change in this regard (but probably one we tolerate).

Yeah that was a point of lengthy discussion in the early days of Razor. There was a proposal in the beginning for a double-`@` syntax for directives (e.g., `@@page`) but in the end single-`@` won, so we do have to accept that all new directives are technically breaking but almost always don't break anyone in practice.